internal/pkg/deploy/pipeline.go

```diff
@@ -124,6 +130,26 @@ func (s *GitHubSource) parseOwnerAndRepo() (owner, repo string, err error) {
 	return matches["owner"], matches["repo"], nil
 }
 
+// parseOwnerAndRepo parses the owner and repo name from the GH repo URL, which was formatted and assigned in cli/pipeline_init.go.
+func (s *GitHubV1Source) parseOwnerAndRepo() (owner, repo string, err error) {
+	if s.RepositoryURL == "" {
```

Comment:
Hmm isn't this very similar to `GitHubSource.parseOwnerAndRepo`? I think maybe we can make `GitHubSource.RepositoryURL` and `GitHubV1Source.RepositoryURL` to be another type lets say `githubURL` with a method `parse()`. We can then simply call `s.RepositoryURL.parse()` to get the owner and repo. Same for the other very similar functions.